src/web/src/app/space/detail/component.ts

```diff
@@ -0,0 +1,277 @@
+	linksPageSize: number = 7;
+	usersPageSize: number = 10;
+
+	invitationsTotal: number;
+	linksTotal: number;
+	usersTotal: number;
+
+	@ViewChild('invitationsPaginator') invitationsPaginator: MatPaginator;
+	@ViewChild('linksPaginator') linksPaginator: MatPaginator;
+	@ViewChild('usersPaginator') usersPaginator: MatPaginator;
+
+	currentSpaceLoaded = false;
+
+	constructor(
+		private route: ActivatedRoute,
+		@Inject(DOCUMENT) private document: any,
+		private fb: FormBuilder,
+		private space: SpaceService,
+		private user: UserService,
+		public requestWatcher: RequestWatcherService,
+	) {
+		this._initInvitationForm();
+		this._initLinkForm();
+	}
+
+	ngOnInit() {
+		let pageSize = localStorage.getItem('invitationsPaginatorSpaceDetailPageSize');
+		this.invitationsPageSize = pageSize ? parseInt(pageSize) : 7;
+		
+		pageSize = localStorage.getItem('linksPaginatorSpaceDetailPageSize');
+		this.linksPageSize = pageSize ? parseInt(pageSize) : 7;
+
+		pageSize = localStorage.getItem('usersPaginatorSpaceDetailPageSize');
+		this.usersPageSize = pageSize ? parseInt(pageSize) : 10;
+
+		this.currentSpaceId = this.route.snapshot.paramMap.get('id');
+		this.getCurrentSpace(this.currentSpaceId);
+		this.getInvitations(this.invitationsPageSize, 0);
+		this.getLinks(this.linksPageSize, 0);
+		this.getUsers(this.usersPageSize, 0);
+	}
+
+	getCurrentSpace(spaceId: string) {
+		this.space.getById(spaceId)
+			.subscribe(res => {
+				this.currentSpace = res;
+				this.currentSpaceLoaded = true;
+			});
+	}
+
+	getInvitations(limit: number, page: number) {
+		this.space.getInvitationListForSpace(this.currentSpaceId, limit, page)
+			.subscribe(res => {
+				this.invitations = res.results;
+				this.invitationsTotal = res.count;
+			});
+	}
+
+	getLinks(limit: number, page: number) {
+		this.space.getLinkListForSpace(this.currentSpaceId, limit, page)
+			.subscribe(res => {
+				this.links = res.results;
+				this.linksTotal = res.count;
+			});
+	}
+
+	getUsers(limit: number, page: number) {
+		this.space.getUserListForSpace(this.currentSpaceId, limit, page)
+			.subscribe(res => {
+				this.users = res.results;
+				this.usersTotal = res.count;
+			});
+	}
+
+	onLinkCopyBtn(link: SpaceLink){
+		let copyToClipboard = this.document.createElement('textarea');
+		copyToClipboard.style.position = 'fixed';
+		copyToClipboard.style.left = '0';
+		copyToClipboard.style.top = '0';
+		copyToClipboard.style.opacity = '0';
+		copyToClipboard.value = this.document.location.origin + this.linksURL + link.id;
+		document.body.appendChild(copyToClipboard);
+		copyToClipboard.focus();
+		copyToClipboard.select();
+		document.execCommand('copy');
+		document.body.removeChild(copyToClipboard);
+	}
+
+	onInvitationDelBtn(invitation: SpaceInvitation){
+		this.space.delInvitationById(invitation.id).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		).subscribe(res => {});
+		this.invitationsPaginator.firstPage();
+		this.getInvitations(this.invitationsPageSize, 0);
+	}
```

Comment:
В текущем виде ты не дожидаясь завершения удаления записи из базы, уже запрашиваешь список. Так тебе может вернуться список в котором удаляемая запись ещё есть.
```suggestion
	onInvitationDelBtn(invitation: SpaceInvitation){
		this.space.delInvitationById(invitation.id).pipe(
			src => this.requestWatcher.WatchFor(src)
		).subscribe(res => {
			this.invitationsPaginator.firstPage();
			this.getInvitations(this.invitationsPageSize, 0);
		});
	}
```

Reply:
Готово (везде).